### ISSUE TITLE
redeclare variable leads to wrong error message

### ISSUE THREAD

**Issue by ovanderzee** [eslint/eslint#13009]
**Tell us about your environment**

Node version: v8.17.0
npm version: v6.13.4
Local ESLint version: v6.8.0 (Currently used)
Global ESLint version: Not found

**What parser (default, Babel-ESLint, etc.) are you using?**
rollup v1.31.0

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

{
    "env": {
        "browser": true,
        "es6": true,
        "jasmine": true,
        "jest/globals": true
    },
    "extends": [
        "eslint:recommended",
        "plugin:jasmine/recommended"
    ],
    "globals": {
        "OpenSeadragon": true
    },
    "parserOptions": {
        "ecmaVersion": 2018,
        "sourceType": "module"
    },
    "plugins": [
        "jasmine",
        "jest"
    ],
    "rules": {
        "no-empty": "warn",
        "no-extra-semi": "warn",
        "no-unused-vars": ["warn"]
    }
}

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

    const table = document.createElement('table')
    let x = 2
    if (table) {
        let x = 10
    }
    console.log('x ' + x)    // 2
    console.log('typeof x ' + typeof x)   // number

then 

    npx eslint {src,test/**}/*.js

**What did you expect to happen?**

I expect an error message for the "disallow variable redeclaration (no-redeclare)" rule

**What actually happened? Please include the actual, raw output from ESLint.**

I got a message pointing me in the wrong direction
    27:13  warning  'x' is assigned a value but never used  no-unused-vars

**Are you willing to submit a pull request to fix this bug?**
no

**Comment by mdjermanovic** [comment]
Hi @ovanderzee, thanks for the issue!

It works as expected. `let` is block-scoped, so there are two different variables named `x`.

`no-redeclare` would catch something like this:

```js
var x = 2
if (table) {
    var x = 10 // 'x' is already defined. (no-redeclare)
}
```

`var` isn't block-scoped, so this is the same variable `x` declared twice.

In your example, the [no-shadow](https://eslint.org/docs/rules/no-shadow) rule would report an error:

```js
/* eslint no-shadow: error */

let x = 2
if (table) {
    let x = 10 //  'x' is already declared in the upper scope. (no-shadow)
}
```

[Online Demo Link](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLXNoYWRvdzogZXJyb3IgKi9cblxubGV0IHggPSAyXG5pZiAodGFibGUpIHtcbiAgICBsZXQgeCA9IDEwXG59Iiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjo2LCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7Im5vLXJlZGVjbGFyZSI6Mn0sImVudiI6e319fQ==)
